lib/ronin/cli/commands/unarchive.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+#
+# Copyright (c) 2006-2023 Hal Brodigan (postmodern.mod3 at gmail.com)
+#
+# Ronin is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ronin is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ronin.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+require 'ronin/cli/file_processor_command'
+require 'ronin/support/archive'
+
+module Ronin
+  class CLI
+    module Commands
+      #
+      # Unarchive the files.
+      #
+      # ## Usage
+      #
+      #     ronin unarchive [options] [FILE ...]
+      #
+      ## Options
+      #
+      #     -f, --format tar|zip             Explicit archive format
+      #
+      # ## Arguments
+      #
+      #     [FILE ...]                       File(s) to unarchive
+      #
+      class Unarchive < FileProcessorCommand
+        ALLOWED_EXTENSIONS  = ['.tar', '.zip']
+        EXTENSIONS_MAPPINGS = {
+          '.tar' => :untar,
+          '.zip' => :unzip
+        }
+        FORMAT_MAPPINGS = {
+          zip: :unzip,
+          tar: :untar
+        }
```

Comment:
I'm not sure if I like that :/

Reply:
You could map `'.tar'` to `:tar` and `'.zip'` to `:zip`, and instead of using `Support::Archive.send` you could do:

```ruby
UNARCHIVE_METHODS = {
  zip: Support::Archive.method(:unzip),
  tar: Support::Archive.method(:untar),
}

...

unarchive_method = UNARCHIVE_METHODS.fetch(options[:format])
unarchive_method.call(...)
```

Or you could do it more explicitly using `case options[:format]`:

```ruby
def run(*files)
  files.each do |file|
    open_archive(file) do |archive|
      # ...
    end
  end
end

def open_archive(file,&block)
  case options[:format]
  when :tar
    Support::Archive.untar(file,&block)
  when :zip
    Support::Archive.unzip(file,&block)  
  end
end
```